mockkubeapiserver/managedfields.go

```diff
@@ -0,0 +1,113 @@
+package mockkubeapiserver
```

Comment:
Yeah, I agree - I'd love to get rid of most of this if we can agree to put it into a library somewhere.  At the same time, figuring out what needs to go into that library is really tough, and generally depending on any kubernetes library that is seen as "internal" is unsupported.  I don't know if `apiserver` is considered something that we wouldn't break just for fun, but I would guess not...

Because this is test code, I propose that I put any code that I "liberate" into a subdirectory called "forked" or something, and then we can see what we would actually like in a library.  Because it's test code, it should be pretty easy also to replace it with the "real" exported function once we have it - if the tests pass, it's good.